tests/certification/pubsub/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,247 @@
+				messages[i] = fmt.Sprintf("testmessageForEventHubCertificationTest #%v", i+1)
+			}
+			messageWatchers.ExpectStrings(messages...)
+			return nil
+		}
+	}
+
+	flow.New(t, "eventhubs certification").
+
+		// Run the subscriberApplication 1
+		Step(app.Run(appID1, fmt.Sprintf(":%d", appPort),
+			subscriberApplication(appID1, topicName1, consumerGroup1))).
+
+		// Run the Dapr sidecar with the eventhubs component 1.
+		Step(sidecar.Run(sidecarName1,
+			embedded.WithComponentsPath("./components/consumer1"),
+
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort),
+			runtime.WithPubSubs(component))).
+
+		// Run the second application.
+		Step(app.Run(appID2, fmt.Sprintf(":%d", appPort+portOffset),
+			subscriberApplication(appID2, topicName1, consumerGroup2))).
+		//
+		// Run the Dapr sidecar with the component 2.
+		Step(sidecar.Run(sidecarName2,
+			embedded.WithComponentsPath("./components/consumer2"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset),
+			runtime.WithPubSubs(component))).
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName1, topicName1, consumerGroup1, consumerGroup2)).
+		Step("publish messages to unUsedTopic", publishMessages(metadata, sidecarName1, unUsedTopic)).
+		Step("Verify if app1 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup1)).
+		Step("Verify if app2 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup2)).
+		Step("reset", flow.Reset(consumerGroup2)).
+
+		// Run the third application.
+		Step(app.Run(appID3, fmt.Sprintf(":%d", appPort+portOffset*2),
+			subscriberApplication(appID3, topicName1, consumerGroup2))).
+
+		// Run the Dapr sidecar with the component 3.
+		Step(sidecar.Run(sidecarName3,
+			embedded.WithComponentsPath("./components/consumer3"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*2),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*2),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*2),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*2),
+			runtime.WithPubSubs(component))).
+
+		// publish message again to topic1, however this time there are more than one consumer group on same topic
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName1, topicName1, consumerGroup2)).
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName2, topicName1, consumerGroup2)).
+		Step("Verify if app1, app3 together has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup2)).
```

Comment:
quick question ... if only a single partition is used, how can two consumers consume from the same partition here? 
```
metadata := map[string]string{
 		messageKey: partition0,
 	}
```

Reply:
Good catch, currently `package watcher` is agnostic of number of owners, hence the test passes.

It's a miss, will publish for one more set of partition key.

Reply:
Fixed.